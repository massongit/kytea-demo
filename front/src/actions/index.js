import * as types from "./types"
import {createAction} from "redux-actions"
import {intl} from "../intl"

/**
 * KyTeaによる解析結果の表示ActionのActionCreator
 * @type {function}
 */
export const showSentence = createAction(types.SHOW_SENTENCE, parameter => {
    // KyTeaによる解析結果の表示Action
    const action = {
        words: []
    }

    // 空白を除去したsentenceをActionにセット
    if (parameter.sentence) {
        action.sentence = parameter.sentence.replace(/\s/g, "")
    }

    if (parameter.words) {
        // 予測不能であった場合のタグ
        const unknownTag = "(Unknown)"

        for (const word of parameter.words) {
            if (word) {
                // Actionのword
                const word_ = {
                    word: word.word,
                    pronunciation: []
                }

                if (word.pos === unknownTag) {
                    word.pos = intl.formatMessage({
                        id: "unknown"
                    })
                } else {
                    word_.pos = word.pos
                }

                for (let i = 0; word.pronunciation && i < word.pronunciation.length; i++) {
                    if (word.pronunciation[i].pronunciation === unknownTag) {
                        word.pronunciation[i].pronunciation = intl.formatMessage({
                            id: "unknown"
                        })
                    }

                    word_.pronunciation.push(word.pronunciation[i])
                }

                action.words.push(word_)
            } else {
                action.words.push(word)
            }
        }
    }

    return action
})

/**
 * 品詞や読みの表示ActionのActionCreator
 * @type {function}
 */
export const showPOSAndPronunciation = createAction(types.SHOW_PRONUNCIATION, parameter => ({
    number: parameter.number,
    word: parameter.word,
    pos: parameter.pos,
    pronunciation: parameter.pronunciation
}))
