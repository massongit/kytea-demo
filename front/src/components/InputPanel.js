import React from "react"
import Input from "../containers/Input"
import {Panel, Row} from "react-bootstrap"
import {FormattedMessage} from "react-intl"

/**
 * 入力部のパネル
 * @returns {element} 入力部のパネル
 */
const InputPanel = () => (
    <Row>
        <Panel>
            <Panel.Heading>
                <FormattedMessage id="title.input"/>
            </Panel.Heading>
            <Panel.Body>
                <p>
                    <FormattedMessage id="description.input"/>
                </p>
                <Input/>
            </Panel.Body>
        </Panel>
    </Row>
)

export default InputPanel
