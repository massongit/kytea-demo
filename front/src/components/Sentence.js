import React from "react"
import PropTypes from "prop-types"
import Word from "../containers/Word"
import {ButtonGroup} from "react-bootstrap"

/**
 * 入力文をボタンのグループとして表示するComponent
 */
class Sentence extends React.Component {
    static propTypes = {
        words: PropTypes.arrayOf(PropTypes.shape({
            pos: PropTypes.string.isRequired,
            word: PropTypes.string.isRequired,
            pronunciation: PropTypes.arrayOf(PropTypes.shape({
                id: PropTypes.number,
                margin: PropTypes.number.isRequired,
                pronunciation: PropTypes.string.isRequired
            })).isRequired
        }).isRequired)
    }

    render() {
        if (this.props.words) {
            return (
                <ButtonGroup>
                    {
                        this.props.words.map((word, i) => (
                            <Word
                                key={i}
                                word={word.word}
                                pos={word.pos}
                                pronunciation={word.pronunciation}
                            />
                        ))
                    }
                </ButtonGroup>
            )
        }
    }
}

export default Sentence
