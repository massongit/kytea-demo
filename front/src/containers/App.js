import {connect} from "react-redux"
import App from "../components/App"

const mapStateToProps = state => ({
    words: state.showSentence.words
})

/**
 * ルート要素のContainer
 */
export default connect(mapStateToProps)(App)
