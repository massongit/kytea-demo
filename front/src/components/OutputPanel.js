import React from "react"
import PropTypes from "prop-types"
import {Panel, Row} from "react-bootstrap"
import Information from "../containers/Information"
import Sentence from "../containers/Sentence"

/**
 * 解析結果表示部のパネル
 * @param word {string} 単語
 * @returns {element} 解析結果表示部のパネル
 */
const OutputPanel = ({word}) => (
    <Row>
        <Panel>
            <Panel.Heading>
                結果
            </Panel.Heading>
            <Panel.Body>
                <p>
                    単語をクリックすると、その単語の品詞や読みが表示されます。
                </p>
                <Sentence/>
            </Panel.Body>
            {
                (() => {
                    // 表示する単語が選択されていない場合には表示しない
                    if (word) {
                        return (<Information/>)
                    }
                })()
            }
        </Panel>
    </Row>
)

OutputPanel.propTypes = {
    word: PropTypes.string
}

export default OutputPanel
