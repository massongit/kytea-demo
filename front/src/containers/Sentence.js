import {connect} from "react-redux"
import Sentence from "../components/Sentence"

const mapStateToProps = state => ({
    words: state.showSentence.words
})

/**
 * 入力文をボタンのグループとして表示するComponentのContainer
 */
export default connect(mapStateToProps)(Sentence)
