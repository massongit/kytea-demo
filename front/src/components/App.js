import React from "react"
import PropTypes from "prop-types"
import Header from "./Header"
import InputPanel from "./InputPanel"
import DescriptionPanel from "./DescriptionPanel"
import OutputPanel from "../containers/OutputPanel"
import {Grid} from "react-bootstrap"

/**
 * ルート要素
 */
class App extends React.Component {
    static propTypes = {
        words: PropTypes.arrayOf(PropTypes.shape({
            pos: PropTypes.string.isRequired,
            word: PropTypes.string.isRequired,
            pronunciation: PropTypes.arrayOf(PropTypes.shape({
                id: PropTypes.number,
                margin: PropTypes.number.isRequired,
                pronunciation: PropTypes.string.isRequired
            })).isRequired
        }).isRequired)
    }

    render() {
        return (
            <Grid>
                <Header/>
                <DescriptionPanel/>
                <InputPanel/>
                {
                    (() => {
                        // KyTeaによる解析が行われていない場合には表示しない
                        if (this.props.words && 0 < this.props.words.length) {
                            return (<OutputPanel/>)
                        }
                    })()
                }
            </Grid>
        )
    }
}

export default App
