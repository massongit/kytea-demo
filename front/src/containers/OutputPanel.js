import {connect} from "react-redux"
import OutputPanel from "../components/OutputPanel"

const mapStateToProps = state => ({
    word: state.showPOSAndPronunciation.word
})

/**
 * 解析結果表示部のパネルのContainer
 */
export default connect(mapStateToProps)(OutputPanel)
