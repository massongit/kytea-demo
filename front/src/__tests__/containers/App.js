import React from "react"
import App from "../../containers/App"
import OutputPanel from "../../containers/OutputPanel"
import InputPanel from "../../components/InputPanel"
import deepcopy from "deepcopy"
import rootReducer from "../../reducers"
import {shallow} from "enzyme"
import {createStore} from "redux"
import {showSentence} from "../../actions"
import {showSentenceParameter} from "../reducers"
import {makeStore} from "./Sentence"

let appComponent

describe("containers/PanelBody/initialState", () => {
    beforeEach(() => {
        appComponent = shallow(
            <App
                store={createStore(rootReducer)}
            />
        ).dive()
    })

    it("初期状態において、Componentが正しく配置されている", () => {
        expect(appComponent).toMatchSnapshot()
    })

    it("初期状態において、子要素にInputPanelが含まれる", () => {
        expect(appComponent.children().contains(<InputPanel/>)).toBeTruthy()
    })

    it("初期状態において、子要素にOutputPanelが含まれない", () => {
        expect(appComponent.children().contains(<OutputPanel/>)).toBeFalsy()
    })
})

describe("containers/PanelBody/showSentenceState", () => {
    beforeEach(() => {
        const store = createStore(rootReducer)
        store.dispatch(showSentence(deepcopy(showSentenceParameter)))
        appComponent = shallow(
            <App
                store={store}
            />
        ).dive()
    })

    it("初期状態からshowSentenceへStateが遷移した際に、Componentが正しく配置されている", () => {
        expect(appComponent).toMatchSnapshot()
    })

    it("初期状態からshowSentenceへStateが遷移した際に、子要素にInputPanelが含まれる", () => {
        expect(appComponent.children().contains(<InputPanel/>)).toBeTruthy()
    })

    it("初期状態からshowSentenceへStateが遷移した際に、子要素にOutputPanelが含まれる", () => {
        expect(appComponent.children().contains(<OutputPanel/>)).toBeTruthy()
    })
})

describe("containers/PanelBody/showPOSAndPronunciationState", () => {
    beforeEach(() => {
        appComponent = shallow(
            <App
                store={makeStore()}
            />
        ).dive()
    })

    it("初期状態からshowSentence, showPOSAndPronunciationとStateが遷移した際に、Componentが正しく配置されている", () => {
        expect(appComponent).toMatchSnapshot()
    })

    it("初期状態からshowSentence, showPOSAndPronunciationとStateが遷移した際に、子要素にInputPanelが含まれる", () => {
        expect(appComponent.children().contains(<InputPanel/>)).toBeTruthy()
    })

    it("初期状態からshowSentence, showPOSAndPronunciationとStateが遷移した際に、子要素にOutputPanelが含まれる", () => {
        expect(appComponent.children().contains(<OutputPanel/>)).toBeTruthy()
    })
})
