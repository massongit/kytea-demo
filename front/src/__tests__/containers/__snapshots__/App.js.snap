// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/App/initialState 初期状態において、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          }
    }
    words={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <Header />,
        <DescriptionPanel />,
        <InputPanel />,
        undefined,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Header />,
          <DescriptionPanel />,
          <InputPanel />,
          undefined,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`containers/App/showPOSAndPronunciationState 初期状態からshowSentence, showPOSAndPronunciationとStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          }
    }
    words={
        Array [
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "やきゅう",
                },
              ],
              "word": "野球",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 2.12891303174829,
                  "pronunciation": "の",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "きの",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "ゅの",
                },
              ],
              "word": "の",
            },
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "ＤＨ",
                },
              ],
              "word": "ＤＨ",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 2.1760714835829633,
                  "pronunciation": "の",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "きの",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "ゅの",
                },
              ],
              "word": "の",
            },
            Object {
              "pos": "形状詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "せいしき",
                },
              ],
              "word": "正式",
            },
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "よびな",
                },
              ],
              "word": "呼び名",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "と",
                },
              ],
              "word": "と",
            },
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "よみ",
                },
              ],
              "word": "読み",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 1.9998295447276375,
                  "pronunciation": "を",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "みを",
                },
                Object {
                  "margin": -0.00005912806885244315,
                  "pronunciation": "くを",
                },
              ],
              "word": "を",
            },
            Object {
              "pos": "動詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "おしえ",
                },
              ],
              "word": "教え",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "て",
                },
              ],
              "word": "て",
            },
            Object {
              "pos": "動詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "くださ",
                },
              ],
              "word": "くださ",
            },
            Object {
              "pos": "語尾",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "い",
                },
              ],
              "word": "い",
            },
            Object {
              "pos": "補助記号",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "。",
                },
              ],
              "word": "。",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <Header />,
        <DescriptionPanel />,
        <InputPanel />,
        <Connect(OutputPanel) />,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Header />,
          <DescriptionPanel />,
          <InputPanel />,
          <Connect(OutputPanel) />,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`containers/App/showSentenceState 初期状態からshowSentenceへStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          }
    }
    words={
        Array [
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "やきゅう",
                },
              ],
              "word": "野球",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 2.12891303174829,
                  "pronunciation": "の",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "きの",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "ゅの",
                },
              ],
              "word": "の",
            },
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "ＤＨ",
                },
              ],
              "word": "ＤＨ",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 2.1760714835829633,
                  "pronunciation": "の",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "きの",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "ゅの",
                },
              ],
              "word": "の",
            },
            Object {
              "pos": "形状詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "せいしき",
                },
              ],
              "word": "正式",
            },
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "よびな",
                },
              ],
              "word": "呼び名",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "と",
                },
              ],
              "word": "と",
            },
            Object {
              "pos": "名詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "よみ",
                },
              ],
              "word": "読み",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 1.9998295447276375,
                  "pronunciation": "を",
                },
                Object {
                  "margin": 0,
                  "pronunciation": "みを",
                },
                Object {
                  "margin": -0.00005912806885244315,
                  "pronunciation": "くを",
                },
              ],
              "word": "を",
            },
            Object {
              "pos": "動詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "おしえ",
                },
              ],
              "word": "教え",
            },
            Object {
              "pos": "助詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "て",
                },
              ],
              "word": "て",
            },
            Object {
              "pos": "動詞",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "くださ",
                },
              ],
              "word": "くださ",
            },
            Object {
              "pos": "語尾",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "い",
                },
              ],
              "word": "い",
            },
            Object {
              "pos": "補助記号",
              "pronunciation": Array [
                Object {
                  "margin": 100,
                  "pronunciation": "。",
                },
              ],
              "word": "。",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <Header />,
        <DescriptionPanel />,
        <InputPanel />,
        <Connect(OutputPanel) />,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Header />,
          <DescriptionPanel />,
          <InputPanel />,
          <Connect(OutputPanel) />,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
